### DISCUSSION TITLE
Topic Size Formula

### DISCUSSION THREAD

**Discussion by n-i-c-o-l-a-s** [tchiotludo/akhq#916] [Q&A] [answered]
Hello. What is the formula to calculate the topic size please ?

We have some Spring Boot app. who use org.apache.kafka.clients.consumer.ConsumerRecord#serializedKeySize and org.apache.kafka.clients.consumer.ConsumerRecord#serializedValueSize : the topic total size at the end is a bit different.

Thank you.

**Comment by tchiotludo** [comment]
To be sure of which size you talk about, can you share a screenshot please ?

**Comment by n-i-c-o-l-a-s** [comment]
<img width="796" alt="topicSize" src="https://user-images.githubusercontent.com/44503667/142866666-e98414ab-023e-42a1-890a-10b8c33619c3.PNG">

**Comment by tchiotludo** [comment, marked as answer]
It's the size reported directly by Kafka (the broker) for all the partition

**Reply by n-i-c-o-l-a-s** [reply]
Ok. I assume you get the value programmatically from your source code: what class and method is in charge of this please ?

**Reply by tchiotludo** [reply]
this one : https://github.com/tchiotludo/akhq/blob/eb330a96322f23feb8096b2ad3f15334c315987e/src/main/java/org/akhq/repositories/LogDirRepository.java#L25

**Comment by n-i-c-o-l-a-s** [comment]
Thank you, Ludo.
When we see 56.032 KB: is it 56 KB or 56032 KB which is about 56MB please ?
Thank you
![image](https://user-images.githubusercontent.com/44503667/146238017-ec93f9a5-9604-4e6a-8a3c-fe79783f31f0.png)

**Reply by tchiotludo** [reply]
it mean to be 56 KB
